gcsfs/core.py

```diff
@@ -1086,54 +1086,70 @@ async def _isdir(self, path):
     async def _find(self, path, withdirs=False, detail=False, prefix="", **kwargs):
         path = self._strip_protocol(path)
         bucket, key = self.split_path(path)
-        out, _ = await self._do_list_objects(
-            path,
-            delimiter=None,
-            prefix=prefix,
-        )
-        if not prefix and not out and key:
-            try:
-                out = [
-                    await self._get_object(
```

Comment:
Got rid of the `_get_object` call in order to only use `_do_list_objects`: by properly providing `prefix`, we can list all objects and then process those

Reply:
Is there a chance that the user has get permission but not list? Is there a performance difference? I ask because similar things came up in s3fs.

Reply:
I guess that could happen as list requires `storage.objects.list` (https://cloud.google.com/storage/docs/access-control/iam-permissions#object_permissions, although I wouldn't expect find (or `ls` or anything that involves `recursive`) to work in this case, but I might have not understood what you mean here.

About performance: when running `find` on exact paths (e.g.: `gs://bucket/folder/file.txt`), we would be calling `self._do_list_objects`:
https://github.com/fsspec/gcsfs/blob/f169fddbb0ace94418d806ba3a5c0706c7dd08db/gcsfs/core.py#L513-L517

which would add a trailing slash to the provided path, thus querying for a non-existent prefix and resulted in having to do an extra call to `_get_object` in order to get the object, attempting to use dircache first and eventually falling back to the same list call that `_do_list_object` uses: overall I think we'll be making less calls.